compiler/z/codegen/S390GenerateInstructions.cpp

```diff
@@ -1011,9 +1011,34 @@ generateRILInstruction(TR::CodeGenerator * cg, TR::InstOpCode::Mnemonic op, TR::
    return new (INSN_HEAP) TR::S390RILInstruction(op, n, treg, label, cg);
    }
 
-TR::Instruction *
-generateRILInstruction(TR::CodeGenerator * cg, TR::InstOpCode::Mnemonic op, TR::Node * n, TR::Register * treg, uintptrj_t imm, TR::Instruction * preced)
+TR::Instruction * // this overload is only for immediate values which are used directly in the instruction and not part of an address calculation
+generateRILInstruction(TR::CodeGenerator * cg, TR::InstOpCode::Mnemonic op, TR::Node * n, TR::Register * treg, uint32_t imm, TR::Instruction * preced)
+   {
+   // this if() makes no sense. The TR_DisableHighWordRA and assignToHPR are contradictory. Maybe the author meant to have '!' in front of the disable?
+   // TODO add an assert to check if this if() is ever entered
```

Comment:
This code looks dead. There are currently no callers in any downstream project which call `generateRILInstruction` with `TR::InstOpCode::AFI` as the mnemonic. Besides upgrading to HPRs is already done in:

https://github.com/eclipse/omr/blob/381ec5b6fb6454f94d78a12b4349c32095e016a2/compiler/z/codegen/OMRCodeGenerator.cpp#L1950-L1951

We should remove the TODO and remove this `if` block.